staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -147,3 +151,33 @@ func ComputeHistogramDelta(before, after model.Samples, label model.LabelName) {
 func makeKey(a, b model.LabelValue) string {
 	return string(a) + "___" + string(b)
 }
+
+// GetGaugeMetricValue extract metric value from GaugeMetric
+func GetGaugeMetricValue(t *testing.T, m metrics.GaugeMetric) float64 {
```

Comment:
I'd like `func GetGaugeMetricValue(m metrics.GaugeMetric)`, not involve testing package. They may be used by integration or E2E in the future.

Same as below

Reply:
I'm not sure if I understand. I put this function in `testutil` because it could be used by tests (unit, integration, e2e) in future

Reply:
Yeah, exactly, It may be re-used by integration and e2e, so it shouldn't take `t *testing.T` parameter.